pkg/activator/handler/handler_test.go

```diff
@@ -419,34 +403,49 @@ func getHandler(throttler *activator.Throttler, t *testing.T, act activator.Acti
 	return handler
 }
 
-func assertResponses(wantedSuccess, wantedFailure, overallRequests int, respCh chan *httptest.ResponseRecorder, t *testing.T) {
+func assertResponses(wantedSuccess, wantedFailure, overallRequests int, lockerCh chan struct{}, respCh chan *httptest.ResponseRecorder, t *testing.T) {
 	t.Helper()
+
+	successCode := http.StatusOK
+	failureCode := http.StatusServiceUnavailable
+
 	var succeeded int
 	var failed int
-	wantedOverflowMessage := activator.ErrActivatorOverload.Error() + "\n"
+
 	for i := 0; i < overallRequests; i++ {
+		if i < wantedSuccess {
```

Comment:
What if we got more requests that reached the round tripper then `wantedSuccess`? Then we would be probably blocked in the next lines waiting for the `respCh`. Should we add a timeout for waiting for `respCh`?

Reply:
Good call, added timeout handling for both of these reads.